services/tenant-ui/frontend/src/components/common/RowExpandData.vue

```diff
@@ -2,32 +2,45 @@
   <div v-if="loading" class="flex justify-content-center">
     <ProgressSpinner />
   </div>
-  <div v-if="data">
+  <div v-if="item">
     <div>
-      <slot name="details" v-bind="data"></slot>
+      <slot name="details" v-bind="item"></slot>
     </div>
     <Accordion>
       <AccordionTab header="View Raw Content">
-        <vue-json-pretty :data="data" />
+        <vue-json-pretty :data="item" />
       </AccordionTab>
     </Accordion>
   </div>
 </template>
 
-<script setup lang="ts">
-import { PropType } from 'vue';
+<script async setup lang="ts">
 import Accordion from 'primevue/accordion';
 import AccordionTab from 'primevue/accordiontab';
 import ProgressSpinner from 'primevue/progressspinner';
 import VueJsonPretty from 'vue-json-pretty';
+import useGetItem from '@/composables/useGetItem';
 
 const props = defineProps({
-  data: {
-    type: null as unknown as PropType<any>,
+  url: {
+    type: String,
     required: true,
   },
-  loading: {
-    type: Boolean,
+  id: {
+    type: String,
+    required: true,
+  },
+  params: {
+    type: Object,
+    required: false,
+    default() {
+      return {};
+    },
   },
 });
+
+const { loading, item, fetchItem } = useGetItem(props.url);
+
+// ok, let's load up the row with acapy data...
+fetchItem(props.id, props.params);
```

Comment:
I'd normally be doing a fetch on an onMounted lifecycle event (or a watched prop or something). I might have just missed it in Vue3 but is this a typical pattern? Loading just as part of the setup script? It seems to be working.

Reply:
Got this pattern when I was looking into composables and loading data asynchronously. The setup script is async, so it'll render when ready.

Reply:
https://vuejs.org/api/sfc-script-setup.html#top-level-await